### PR TITLE
Add STM32WL support

### PR DESCRIPTION
**Pull request by ABOSTM**
Add STM32WL support:
* STM32 HAL/LL
* common_ll
* pinctrl

### CODE REVIEW

File: scripts/genpinctrl/genpinctrl.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "stm32l5": 0x42020000,
     "stm32mp1": 0x50002000,
     "stm32wb": 0x48000000,
+    "stm32wl": 0x48000000,
```
Comment (gmarull) [inline scripts/genpinctrl/genpinctrl.py:67]: This change should go into a separate commit. In order to make reviews easier, autogenerated content should go into separate commits.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: scripts/genpinctrl/genpinctrl.py (hunk 1 above)
Comment (ABOSTM) [reply scripts/genpinctrl/genpinctrl.py:67]: OK. Done

**Review by gmarull — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
Is there a Zephyr side PR associated with this?

**Comment by erwango** [comment]
@galak https://github.com/zephyrproject-rtos/zephyr/pull/32169

**Comment by galak** [comment]
Was waiting to see a CI run of #32169 pass before merging.
